src/test/shell/bazel/remote/build_without_the_bytes_test.sh

```diff
@@ -2010,7 +2010,138 @@ EOF
       --experimental_remote_cache_eviction_retries=1 \
       //a:bar >& $TEST_log || fail "Failed to build"
 
-  expect_log 'Failed to fetch blobs because they do not exist remotely.'
+  expect_log 'lost inputs with digests:'
+  expect_log "Found transient remote cache error, retrying the build..."
+
+  local invocation_ids=$(grep "Invocation ID:" $TEST_log)
+  local first_id=$(echo "$invocation_ids" | head -n 1)
+  local second_id=$(echo "$invocation_ids" | tail -n 1)
+  if [ "$first_id" == "$second_id" ]; then
+    fail "Invocation IDs are the same"
+  fi
+}
+
+# This test documents the current behavior, not the desired behavior.
+function test_remote_cache_eviction_retries_toplevel_artifact() {
+  mkdir -p a
+
+  cat > a/BUILD <<'EOF'
+genrule(
+  name = 'foo',
+  srcs = ['foo.in'],
+  outs = ['foo.out'],
+  cmd = 'cat $(SRCS) > $@',
+)
+EOF
+
+  echo foo > a/foo.in
+
+  # Populate remote cache
+  bazel build \
+      --remote_executor=grpc://localhost:${worker_port} \
+      --remote_download_minimal \
+      //a:foo >& $TEST_log || fail "Failed to build"
+
+  bazel clean
+
+  # Clean build, foo.out isn't downloaded
+  bazel build \
+      --remote_executor=grpc://localhost:${worker_port} \
+      --remote_download_minimal \
+      //a:foo >& $TEST_log || fail "Failed to build"
+
+  if [[ -f bazel-bin/a/foo.out ]]; then
+    fail "Expected top-level output bazel-bin/a/foo.out to not be downloaded"
+  fi
+
+  # Evict blobs from remote cache
+  stop_worker
+  start_worker
+
+  # Incremental build in toplevel build triggers remote cache eviction error
+  # but Bazel doesn't automatically retry the build yet.
+  # TODO: This documents the current behavior, but it's not intended.
```

Comment:
@justinhorvitz To fix this, I would need to thread information about lost inputs obtained in https://github.com/bazelbuild/bazel/blob/998e7624093422bee06e65965f8a575d05d57c27/src/main/java/com/google/devtools/build/lib/skyframe/CompletionFunction.java#L375 through to the `ImportantOutputHandler`. I think I roughly understand what that would require, but the two calls to `informImportantOutputHandler` further above and the mentioning of error bubbling tell me that this is probably pretty difficult to get right. Do you have any advice for me?